src/clj/fluree/db/query/sparql/translator.cljc

```diff
@@ -364,17 +364,18 @@
   ;; NamedGraphClause ::= <'NAMED'> SourceSelector
   ;; <SourceSelector> ::= iri
   [[_ source]]
-  (throw (ex-info "FROM NAMED is not a supported SPARQL clause"
-                  {:status 400 :error :db/invalid-query}))
   (parse-term source))
 
 (defmethod parse-rule :DatasetClause
-  ;; DatasetClause ::= <'FROM'> WS ( DefaultGraphClause | NamedGraphClause )
-  ;; DefaultGraphClause ::= SourceSelector
-  ;; NamedGraphClause ::= <'NAMED'> SourceSelector
-  ;; <SourceSelector> ::= iri
-  [[_ source]]
-  [[:from (parse-term source)]])
+  ;; DatasetClause ::= FromClause*
+  ;; FromClause   ::=   <'FROM'> WS ( DefaultGraphClause | NamedGraphClause )
```

Comment:
Can you add angle brackets around the left hand side FromClause term to match the bnf file?